src/Context/filter-context.js

```diff
@@ -0,0 +1,57 @@
+import { createContext, useReducer, useContext} from "react";
+
+const FilterContext = createContext(null);
+
+
+const FilterReducerFun = (state, action) => {
+  switch (action.type) {
+    //   Sorting
+    case "lowtoHigh":
+      return { ...state, sortby: "ltoh" };
+    case "Hightolow":
+      return { ...state, sortby: "htol" };
+    case "new_arrival":
+      return { ...state, sortby: "new" };
+    case "with_discount":
+      return { ...state, sortby: "discount" };
```

Comment:
Well, it's not necessary but could we maintain the case strings in uppercase with underscores? i.e. `LOW_TO_HIGH`, `HIGH_TO_LOW`, etc. As it will ease reading & also help differentiate these actions types in our code base.